101-vm-ubuntu-DSVM-GPU-or-CPU/azuredeploy.json

```diff
@@ -0,0 +1,311 @@
+            "metadata": {
+                "description": "Password for for Administrator Account"
+            }
+        },
+        "vmName": {
+            "type": "string",
+            "defaultValue": "[concat('DataScienceVM-', uniqueString(utcNow('F')))]",
+            "metadata": {
+                "description": "The name of you Virtual Machine."
+            }
+        },
+        "location": {
+            "type": "string",  
+            "defaultValue": "[resourceGroup().location]",  
+            "metadata": {  
+                    "description": "Location for all resources."  
+            }  
+        },
+        "Cpu-Gpu": {
+            "type": "string",
+            "defaultValue": "cpu",
+            "allowedValues": [
+                "cpu",
+                "gpu"
+            ],
+            "metadata": {
+                "description": "Choose between CPU or GPU processing"
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "vNet",
+            "metadata": {
+                "description": "Name of the VNET"
+            }
+        },
+        "subnetName": {
+            "type": "string",
+            "defaultValue": "subnet",
+            "metadata": {
+                 "description": "Name of the subnet in the virtual network"
+            }
+        },
+        "networkSecurityGroupName": {
+            "type": "string",
+            "defaultValue": "SecGroupNet",
+            "metadata": {
+                "description": "Name of the Network Security Group"
+            }
+        },
+        "virtualMachineSizeCPU": {
+            "type": "string",
+            "defaultValue": "Standard_DS1_v2",
+            "allowedValues": [
+                "Standard_DS1_v2"
+            ],
+            "metadata":{
+                "description": "CPU VM size."
+            }
+        },
+        "virtualMachineSizeGPU": {
+            "type": "string",
+            "defaultValue": "Standard_NC6_Promo",
+            "allowedValues": ["Standard_NC6_Promo"],
+            "metadata":{
+                "description": "GPU VM size."
+            }
+        }
+    },
+    "variables": {
+        "networkInterfaceName": "[concat(parameters('vmName'),'NetInt')]",
+        "virtualMachineName": "[parameters('vmName')]",
+        "publicIpAddressName": "[concat(parameters('vmName'), 'PublicIP' )]",
+        "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetworkName'), parameters('subnetName'))]",
+        "nsgId": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('networkSecurityGroupName'))]",
+        "osDiskType": "Standard_LRS",
+        "storageAccountName": "[concat('storage', uniqueString(resourceGroup().id))]",
+        "storageAccountType": "Standard_LRS",
+        "storageAccountKind": "Storage"
+    },
+    "resources": [
+        {
+            "name": "[variables('networkInterfaceName')]",
```

Comment:
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties

Reply:
Sorted properties